addOns/allinonenotes/src/main/java/org/zaproxy/zap/extension/allinonenotes/ExtensionAllInOneNotes.java

```diff
@@ -0,0 +1,283 @@
+        if (getView() != null) {
+
+            extensionHook.getHookMenu().addToolsMenuItem(getMenuReload());
+            hookView = extensionHook.getHookView();
+            hookView.addStatusPanel(getStatusPanel());
+        }
+    }
+
+    @Override
+    public boolean canUnload() {
+        // The extension can be dynamically unloaded, all resources used/added can be freed/removed
+        // from core.
+        return true;
+    }
+
+    @Override
+    public void unload() {
+        super.unload();
+        // Unloading the event consumer
+        ZAP.getEventBus()
+                .unregisterConsumer(
+                        eventConsumerImpl,
+                        HistoryReferenceEventPublisher.getPublisher().getPublisherName());
+    }
+
+    @Override
+    public void sessionChanged(final Session session) {
+        // session changed - extension should basically act as if "Reload" has been called and
+        // re-build the status panel
+        resetNotesTable();
+    }
+
+    @Override
+    public void sessionAboutToChange(Session session) {
+        // Left empty for now
+    }
+
+    @Override
+    public void sessionScopeChanged(Session session) {
+        // Left empty for now
+    }
+
+    @Override
+    public void sessionModeChanged(Control.Mode mode) {
+        // Left empty for now
+    }
+
+    private ExtensionHistory getExtensionHistory() {
+        if (extHistory == null) {
+            extHistory =
+                    org.parosproxy.paros.control.Control.getSingleton()
+                            .getExtensionLoader()
+                            .getExtension(ExtensionHistory.class);
+        }
+        return extHistory;
+    }
+
+    private AbstractPanel getStatusPanel() {
+
+        if (statusPanel == null) {
+            statusPanel = new AbstractPanel();
+            statusPanel.setLayout(new BorderLayout());
+            statusPanel.setName(Constant.messages.getString(PREFIX + ".panel.title"));
+            statusPanel.setIcon(ICON);
+            statusPanel.add(getToolBar(), BorderLayout.NORTH);
+            statusPanel.add(new JScrollPane(getNotesTable()), BorderLayout.CENTER);
+        }
+
+        return statusPanel;
+    }
+
+    private NotesTableModel getNotesTableModel() {
+        if (notesTableModel == null) {
+            notesTableModel = new NotesTableModel();
+        }
+        return notesTableModel;
+    }
+
+    private JXTable getNotesTable() {
+        if (notesTable == null) {
+            notesTable = new JXTable();
+            notesTable.setModel(getNotesTableModel());
+            notesTable.setColumnSelectionAllowed(false);
+            notesTable.setCellSelectionEnabled(false);
+            notesTable.setRowSelectionAllowed(true);
+            notesTable.setAutoCreateRowSorter(true);
+            notesTable.setColumnControlVisible(true);
+            // TODO: These two options should probably be allowed
+            // Perhaps when this is promoted to beta
```

Comment:
Why just when beta and not now?

Reply:
Because I was trying to move it along.... we can discuss the necessary change effort.

Reply:
It seems that along the way we've addressed the changes necessary to prevent the exceptions I had been encountering when things were sorted/reordered. \o/